tools/CheckH5.py

```diff
@@ -0,0 +1,207 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import unittest
+import numpy as np
+import sys
+import yaml
+import os
+import h5py
+
+
+# Describes a particular comparison between H5 datasets or groups
+class H5Check:
+    # `unit_test` must be the unit test object -- this class is to be
+    # constructed within a test case, and must have access to the
+    # calling unit test object so that it can access the assert and
+    # subTest member functions.
```

Comment:
could you document all the arguments? I don't know what `test_h5_entity` should be and `expected_h5_entity`. It appears these differ with what exactly is being tested (a group, dataset or single number)